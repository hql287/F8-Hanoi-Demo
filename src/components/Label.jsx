// React Libraries
import React from 'react';
import styled from 'styled-components';
const LabelStyle = styled.p`
  text-transform: uppercase;
  letter-spacing: 1px;
  font-size: 14px;
  font-family: sans-serif;
  font-weight: 600;
  background: black;
  position: relative;
  display: inline-block;
  padding: 4px 10px;
  margin-left: 20px;
  border-radius: 4px;
  color: white;
`;

const Label = ({text}) => <LabelStyle><p>{text}</p></LabelStyle>;

export default Label;
