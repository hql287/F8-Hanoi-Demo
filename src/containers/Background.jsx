// React Libraries
import React from 'react';
import Label from '../components/Label';
import path from 'path';

const tilePath = path.resolve(__dirname, './static/imgs/tile_small.png');
const pipePath = path.resolve(__dirname, './static/imgs/pipe.png');

import styled from 'styled-components';

export const Layout = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
`;

export const Sky = styled.div`
  flex: 1;
  background: lightblue;
`;

export const Ground = styled.div`
  flex: 1;
  background: url(${tilePath});
  ${props =>
    props.groundHeight &&
    `
    max-height: ${props.groundHeight}px;
  `};
  ${props =>
    props.borderHeight &&
    `
    border-top: ${props.borderHeight}px solid darkgreen;
  `};
`;

function Background(props) {
  return (
    <Layout>
      <Sky>
        { props.showLabels &&
          <Label text="Sky" />
        }
      </Sky>
      <Ground {...props}>
        { props.showLabels &&
          <Label text="Ground" />
        }
      </Ground>
    </Layout>
  );
}

Background.defaultProps = {
  showLabels: false,
  groundHeight: 240,
  borderHeight: 10,
};

export default Background;
