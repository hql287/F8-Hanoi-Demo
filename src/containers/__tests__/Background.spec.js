// Libs
import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

// Component to test
import Label from '../../components/Label';
import Background, { Layout, Sky, Ground } from '../Background';

// Setup
const pos = { x: 0, y: 0 };
const groundHeight = 160;
const borderHeight = 10;

describe('Renders the Background correctly to the DOM', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Background
        showLabels={false}
        groundHeight={groundHeight}
        borderHeight={borderHeight}
      />
    );
  })

  it('should render a Layout component', () => {
    expect(wrapper.find(Layout)).toHaveLength(1);
    expect(wrapper.find(Layout)).not.toHaveLength(2);
  });

  it('should render a Sky component', () => {
    expect(wrapper.find(Sky)).toHaveLength(1);
    expect(wrapper.find(Sky)).not.toHaveLength(2);
  });

  it('should render a Ground component', () => {
    expect(wrapper.find(Ground)).toHaveLength(1);
    expect(wrapper.find(Ground)).not.toHaveLength(2);
  });

  it('should render the components with correct props', () => {
    expect(wrapper.find(Ground).get(0).props.showLabels).toEqual(false);
    expect(wrapper.find(Ground).get(0).props.groundHeight).toEqual(groundHeight);
    expect(wrapper.find(Ground).get(0).props.borderHeight).toEqual(borderHeight);
  })

  it('should hide/show labels correctly', () => {
    expect(wrapper.find(Label)).toHaveLength(0);
    const newWarpper = shallow(
      <Background
        showLabels={true}
        groundHeight={groundHeight}
        borderHeight={borderHeight}
      />
    );
    expect(newWarpper.find(Label)).toHaveLength(2);
  });

  // Test style rules
  it('should render the components with correct style', () => {

  });

  // Snapshot Testing
  it('should match snapShot', () => {
    const tree = renderer
      .create(
        <Background
          showLabels={false}
          pos={pos}
          groundHeight={groundHeight}
          borderHeight={borderHeight}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})
