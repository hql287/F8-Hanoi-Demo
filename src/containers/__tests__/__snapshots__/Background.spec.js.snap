// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the Background correctly to the DOM should match snapShot 1`] = `
<div
  className="Background__Layout-iLolKP cVmhBA"
>
  <div
    className="Background__Sky-jUevNA ljmzoM"
  />
  <div
    className="Background__Ground-kryFUJ hzgNtZ"
  />
</div>
`;
